docs/content/guides/dagster/using-environment-variables-and-secrets.mdx

```diff
@@ -0,0 +1,199 @@
+---
+title: Using environment variables and secrets | Dagster Docs
+description: Learn how to securely define and access environment variables in your Dagster code.
+---
+
+# Using environment variables and secrets
+
+<EnvironmentVariablesIntro />
+
+---
+
+## Declaring environment variables
+
+How environment variables are declared depends on how your Dagster project is deployed.
+
+### Dagster Cloud
+
+Environment variables can be set a variety of ways in Dagster Cloud, including directly in the UI. Refer to the [Dagster Cloud environment variables guide](/dagster-cloud/developing-testing/environment-variables-and-secrets) for more info.
+
+### Dagster Open Source
+
+Setting environment variables in Dagster Open Source depends on where Dagster is deployed. Refer to the deployment guide for your platform for more info:
+
+<ArticleList>
+  <ArticleListItem
+    title="Amazon Web Services EC2 / ECS"
+    href="/deployment/guides/aws#secrets-management-in-ecs"
+  ></ArticleListItem>
+  <ArticleListItem
+    title="Docker"
+    href="/deployment/guides/docker"
+  ></ArticleListItem>
+  <ArticleListItem
+    title="Kubernetes"
+    href="/deployment/guides/kubernetes/customizing-your-deployment#security"
+  ></ArticleListItem>
+</ArticleList>
+
+---
+
+## Syntax
+
+To access a configured environment variable in your Dagster code, use the following syntax:
+
+```python
+"PARAMETER_NAME": {"env": "ENVIRONMENT_VARIABLE_NAME"}
+```
+
+For example:
+
+```python
+"access_token": {"env": "GITHUB_ACCESS_TOKEN"}
+```
+
+And to access [built-in environment variables for Dagster Cloud](/dagster-cloud/developing-testing/environment-variables-and-secrets#built-in-environment-variables), you could do something like the following:
+
+```python
+os.getenv("DAGSTER_CLOUD_DEPLOYMENT_NAME")
+```
+
+---
+
+## Handling secrets
+
+[Configurable Dagster objects](/concepts/configuration/config-schema) - such as ops, assets, resources, I/O managers, and so on - can accept configuration from environment variables.
+
+Using environment variables to provide secrets ensures sensitive info won't be visible in your code or the launchpad in the UI. In Dagster, best practice for handling secrets uses [configuration](/concepts/config-schema) and [resources](/concepts/resources).
```

Comment:
I think it is less that resources are a best practice for secrets than that resources are a best practice for accessing external services (like Github in the example), and to use external services you usually need a secret to authenticate yourself to the service. Maybe nitpicking.